src/components/RecipeForms/RecipeForm.tsx

```diff
@@ -110,6 +110,57 @@ export const RecipeForm: React.FC<RecipeFormProps> = ({
             )}
           />
 
+          <Controller
+            name="thumbnail"
+            control={control}
+            defaultValue={defaultState.thumbnail}
+            rules={{
+              required: 'You have to add thumbnail',
+            }}
+            render={({ field: { onChange, value } }) => (
+              <div>
+                <input
+                  accept="image/*"
+                  style={{ display: 'none' }}
+                  id="thumbnail"
+                  type="file"
+                  onChange={(e) => {
+                    if (e.target.files) {
+                      return onChange(e.target.files[0]);
+                    }
+                    return onChange(e.target.files);
+                  }}
+                />
+                <label
+                  htmlFor="thumbnail"
+                  className="images-upload__button-container"
+                >
+                  <Button fullWidth variant="contained" component="span">
+                    {type === 'create' ? 'Add ' : 'Update '} thumbnail of your
+                    recipe
+                  </Button>
+                </label>
+                {value && (
+                  <div className="images-upload__images-container">
+                    <div
+                      key={value.name}
+                      className="images-upload__image-container"
+                    >
+                      <img
+                        className="images-upload__image"
+                        alt="Preview"
+                        src={value.url || URL.createObjectURL(value)}
```

Comment:
If thumbnail can be null then you have to handle this with question mark:
```suggestion
                        src={value?.url || URL.createObjectURL(value)}
```